### DISCUSSION TITLE
is it possible to force a split buffer to scroll and show the last lines?

### DISCUSSION THREAD

**Discussion by andreacfromtheapp** [MunifTanjim/nui.nvim#327] [Q&A] [answered]
Hi,

the following function adds line to the bottom right split pane/buffer. As you can see, the NuiLines don't scroll. is there a way to force that buffer to scroll please? I tried with the commented out vim lines but nope.

```lua
-- run chuck vm in a job and putput stdout in a split as is
local function start_chuck(cmd, logfile)
  vim.fn.jobstart(cmd .. " 3>&1 2>&1 | tee " .. tostring(logfile), {
    on_stdout = function(_, data, _)
      for _, line in pairs(data) do
        layout.chuck_log:add_node(NuiTree.Node({ log = line }))
        layout.chuck_log:render()
        -- vim.api.nvim_feedkeys("<S-g>", "n", true)
        -- vim.fn.win_execute(layout.chuck_pane.bufnr, "norm <S-g>")
      end
    end,
  })
end
```

Here's a simple session GIF to see what I mean. Don't mind anything happening but the bottom right pane not scrolling. To explain, what I do is feed the buffers to chuck, the top right shows a list of what's "current" and the bottom right shows the vm logging of everything. With new lines "going over the bottom" It should scroll to show latest lines in my ideal world but I'm probably misunderstading NUI again. Is it feasible at all?

![test](https://github.com/MunifTanjim/nui.nvim/assets/3269984/3d92911e-6c98-4565-a527-5504f8f09767)

Please and thank you

**Comment by MunifTanjim** [comment, marked as answer]
Scrolling is outside the scope of `nui.nvim`.

Scrolling in Neovim is just placing the cursor in right place. You can use `vim.api.nvim_win_set_cursor` for that.

Check `:help nvim_win_set_cursor`.

If your cursor is already on the buffer, you can do:

```lua
vim.api.nvim_win_set_cursor(0, { vim.fn.line('$'), 1 })
```

If your cursor is on a different buffer, you can do:

```lua
vim.api.nvim_buf_call(bufnr, function()
  vim.api.nvim_win_set_cursor(0, { vim.fn.line('$'), 1 })
end)
```

`bufnr` should be the buffer number for which you want to change the cursor.

**Reply by andreacfromtheapp** [reply]
Hi @MunifTanjim :)

I'll try that. Thanks :)

ps: the current way I'm feeding the vm to nui isn't ideal but it kinda works for now. In the future I'd like to find a way to better separate concerns as you suggested [here](https://github.com/MunifTanjim/nui.nvim/discussions/326). I haven't managed yet.

**Reply by andreacfromtheapp** [reply]
it works perfectly! thank you sooooo much <3
